File: 760 - Find Anagram Mappings.swift
class Solution {
    func anagramMappings(_ A: [Int], _ B: [Int]) -> [Int] {
        var mapping = [Int: Int]()
        for (index, number) in B.enumerated() {
            mapping[number] = index
        }
        
        return A.map { mapping[$0]! }
    }
}

File: 760 - Find Anagram Mappings.java
class Solution {
    public int[] anagramMappings(int[] A, int[] B) {
        HashMap<Integer, Integer> hashMap = new HashMap<>();
        
        for (int i = 0; i < B.length; i++) {
            hashMap.put(B[i], i);
        }
        
        for (int i = 0; i < A.length; i++) {
            A[i] = hashMap.get(A[i]);            
        }
        
        return A;
    }
}